### ISSUE TITLE
Ability to query all possible values of a dimension

### ISSUE THREAD

**Issue by lucguillemot** [zazuko/query-rdf-data-cube#20]
Thank you for this library, it 's extremely useful!

Currently, I can easily retrieve the dimensions of a datacube with `datacube.dimensions()`. Would it be possible to also retrieve all the possible values that a dimension can have? For instance all the spatial units that a spatial dimension can have. 

A SPARQL query would look something like this I think:
```
SELECT DISTINCT ?value WHERE {
      ?obs a qb:Observation;
        <dimension>/rdfs:label ?value .
    }
```

The goal is to then query the observations with a filter value like this:
```
datacube
  .query()
  .filter(dimension.equals(dimensionValue))
```

**Comment by vhf** [comment]
> Not sure we'll really need this but: the dimension values can actually change based on the query you execute on a dataset (cube?). For example, if you filter observations by an area, the time dimension could have a different range – in the context of this query.
> 
> Now, this behavior would be quite terrible:
> 
> ```js
> await myDimension.values() // => [1,2,3,4,5]
> 
> await cube.query().filter(something).execute() // ...
> 
> await myDimension.values() // => [3,5]
> ```
> 
> because we'd run into all kinds of bugs if the internal state of `myDimension` would somehow change when queries are executed.

Executing a query or generating sparql for a query doesn't modify any state, if it does it's a bug. Same with dimensions/attributes/measures: `myDimension.distinct()` doesn't mutate `myDimension`.

I'll implement something and if the API for getting the values is confusing we'll find a solution that we all like.

**Comment by herrstucki** [comment]
@vhf sorry, I wasn't trying to say that querying actually _did_ mutate anything. I was just giving an example of a potentially bad API :)

**Comment by vhf** [comment]
Getting all values:
```js
const values = await dataCube.query()
  .select({ size: sizeClasses })
  .filter(({ size }) => size.notEquals("50 - 100 ha"))
  .componentValues();

// same results (but not same SPARQL query!) as:

const values = (await dataCube.componentValues(sizeClasses))
  .filter((value) => value.label.value !== "50 - 100 ha");
```

Getting min/max:
```js
// of a Component, no filter:
const timeMinMax = await dataCube.componentMinMax(time);

// more fine grained on Query, with filters:
const { min: sizeMin, max: sizeMax } = await dataCube.query()
  .select({ size: sizeDimension })
  .filter(({ size }) => size.gt(50))
  .filter(({ size }) => size.lte(250))
  .componentMinMax();
```

**Comment by vhf** [comment]
Published as `@zazuko/query-rdf-data-cube@0.0.11`
